File: lib/simple_event_modeling/cart/commands.rb
# frozen_string_literal: true

module SimpleEventModeling
  module Cart
    module Commands
      Unknown = Struct.new(:aggregate_id)
      CreateCart = Struct.new(:aggregate_id)
      AddItem = Struct.new(:aggregate_id, :item_id)
      RemoveItem = Struct.new(:aggregate_id, :item_id)
      ClearCart = Struct.new(:aggregate_id)
    end
  end
end

File: go/gpt5/cart/commands.go
package cart

// Commands are simple records with no behavior.

type CreateCart struct{ AggregateID string }

type AddItem struct {
	AggregateID string
	ItemID      string
}

type RemoveItem struct {
	AggregateID string
	ItemID      string
}

type ClearCart struct{ AggregateID string }
